qualtran/bloqs/basic_gates/rotation.py

```diff
@@ -72,30 +63,50 @@ class ZPowGate(CirqGateAsBloqBase):
     """
 
     exponent: float = 1.0
-    global_shift: float = 0.0
     eps: float = 1e-11
 
     @cached_property
     def cirq_gate(self) -> cirq.Gate:
-        return cirq.ZPowGate(exponent=self.exponent, global_shift=self.global_shift)
+        exponent = self.exponent + (-1) ** np.random.randint(2) * self.eps / np.pi
+        return cirq.ZPowGate(exponent=exponent, global_shift=0)
 
     def __pow__(self, power):
         g = self.cirq_gate**power
-        return ZPowGate(g.exponent, g.global_shift, self.eps)
+        return ZPowGate(g.exponent, self.eps)
+
+    def t_complexity(self) -> 'TComplexity':
+        if cirq.has_stabilizer_effect(self.cirq_gate):
+            return TComplexity(clifford=1)
+        return TComplexity(rotations=1)
 
 
 @frozen
 class CZPowGate(CirqGateAsBloqBase):
     exponent: float = 1.0
-    global_shift: float = 0.0
     eps: float = 1e-11
 
     @cached_property
     def cirq_gate(self) -> cirq.Gate:
-        return cirq.CZPowGate(exponent=self.exponent, global_shift=self.global_shift)
+        exponent = self.exponent + (-1) ** np.random.randint(2) * self.eps / np.pi
+        return cirq.CZPowGate(exponent=exponent)
 
     def _t_complexity_(self) -> 'TComplexity':
-        return TComplexity(rotations=1)
+        if cirq.has_stabilizer_effect(self.cirq_gate):
+            return TComplexity(clifford=1)
+        return TComplexity(rotations=1, t=4, clifford=9 + 4)
+
+    def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
+        # A CZ**k can be implemented via 1 AND/AND† and 1 non-controlled Z**k
+        if cirq.has_stabilizer_effect(self.cirq_gate):
+            return super().build_call_graph(ssa)
```

Comment:
this is very roundabout. What are you trying to achieve here

Reply:
what do you mean "if this node is a leaf"? This is a CZPowGate

Reply:
Do not override the `build_composite_graph` method in this case.